src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedParameterSymbol.cs

```diff
@@ -228,26 +229,34 @@ public override ImmutableArray<CustomModifier> RefCustomModifiers
             get { return ImmutableArray<CustomModifier>.Empty; }
         }
 
-        private sealed class SynthesizedParameterSymbolWithCustomModifiers : SynthesizedParameterSymbolBase
+        private sealed class SynthesizedComplexParameterSymbol : SynthesizedParameterSymbolBase
         {
             private readonly ImmutableArray<CustomModifier> _refCustomModifiers;
+            private readonly ImmutableArray<CSharpAttributeData> _attributes;
 
-            public SynthesizedParameterSymbolWithCustomModifiers(
+            public SynthesizedComplexParameterSymbol(
                 MethodSymbol container,
                 TypeWithAnnotations type,
                 int ordinal,
                 RefKind refKind,
                 string name,
-                ImmutableArray<CustomModifier> refCustomModifiers)
+                ImmutableArray<CustomModifier> refCustomModifiers,
+                ImmutableArray<CSharpAttributeData> attributes)
                 : base(container, type, ordinal, refKind, name)
             {
                 _refCustomModifiers = refCustomModifiers.NullToEmpty();
```

Comment:
>.NullToEmpty() [](start = 56, length = 14)

Is this necessary?

Consider asserting that `refCustomModifiers` and `attributes` are not default values.

Reply:
True, it looks redundant.